src/test/java/teammates/test/driver/TestProperties.java

```diff
@@ -137,6 +143,64 @@ public static String extractVersionNumber(String inputString) {
         return inputString.substring(startPos, endPos).replace("-", ".").trim();
     }
     
-    
+    public void verifyReadyForGodMode() {
+        if (!inst().isDevServer()) {
+            Assumption.fail("God mode regeneration works only in dev server.");
+        }
+        if (areTestAccountsDefaultValues()) {
+            Assumption.fail("Please change ALL the default accounts in test.properties in order to use God mode,"
+                            + "e.g change test.student1.account from alice.tmms to alice.tmms.example");
+        }
+        if (!areAllTestAccountsDifferent()) {
+            Assumption.fail("ALL test accounts used must be different, including after truncation, "
+                            + "e.g veryveryveryveryverylong and veryveryveryveryverylengthy are not "
+                            + "accepted as two different accounts.");
+        }
+        if (!areAppUrlsDifferent()) {
+            Assumption.fail("App URLs defined in test.properties and build.properties must be different, "
+                            + "and neither one can be a substring of the other, e.g localhost:8888 and "
+                            + "localhost:88889 are not accepted as two different app URLs.");
+        }
+    }
+
+    private boolean areTestAccountsDefaultValues() {
+        /*
+         * TODO make this check much, much stricter. If TEST_STUDENT1_ACCOUNT is charlie.tmms and
+         * TEST_STUDENT2_ACCOUNT is alice.tmms, etc, this check will pass BUT the unintended replacement
+         * will still happen.
+         * Also consider adding checks for the accounts used in other UI tests such as
+         * ISR.teammates.test, instructorWith2Courses, etc. If this is done the method might need a new name.
```

Comment:
This comment is not easy to understand and fragile (can get outdated easily). On a related note, a comment is not a 'note-to-self', so need to explain things more carefully so that a different developer can still understands it.